tests/integration/php/Core/Framework/DataAbstractionLayer/Search/Parser/SqlQueryParserTest.php

```diff
@@ -0,0 +1,91 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Tests\Integration\Core\Framework\DataAbstractionLayer\Search\Parser;
+
+use Doctrine\DBAL\Connection;
+use PHPUnit\Framework\TestCase;
+use Shopware\Core\Framework\Context;
+use Shopware\Core\Framework\DataAbstractionLayer\Dbal\EntityDefinitionQueryHelper;
+use Shopware\Core\Framework\DataAbstractionLayer\EntityRepository;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Filter\EqualsFilter;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Parser\ParseResult;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Parser\SqlQueryParser;
+use Shopware\Core\Framework\Test\TestCaseBase\IntegrationTestBehaviour;
+use Shopware\Core\Framework\Test\TestCaseHelper\ReflectionHelper;
+
+/**
+ * @internal
+ *
+ * @covers \Shopware\Core\Framework\DataAbstractionLayer\Search\Parser\SqlQueryParser
+ */
+class SqlQueryParserTest extends TestCase
```

Comment:
In the integration test we should actually check that the SQL works.

So I would use the `\Shopware\Core\Content\Test\Product\ProductBuilder` (check other tests how to use) and create few products and run several Criteria against that products with ListFields and assert the result :)

Reply:
Thanks! I'll make another attempt :)